exporter/collector/metrics.go

```diff
@@ -190,7 +190,17 @@ func (me *MetricsExporter) PushMetrics(ctx context.Context, m pmetric.Metrics) e
 	// they need to be sent to. Each project's timeseries are sent in a
 	// separate request later.
 	pendingTimeSeries := map[string][]*monitoringpb.TimeSeries{}
-	rms := m.ResourceMetrics()
+
+	rms := pmetric.NewResourceMetricsSlice()
+	m.ResourceMetrics().CopyTo(rms)
+
+	// add extra metrics from the ExtraMetrics() extension point, combine into a new copy
+	extraResourceMetrics := pmetric.NewResourceMetricsSlice()
+	if me.cfg.MetricConfig.ExtraMetrics != nil {
+		extraResourceMetrics = me.cfg.MetricConfig.ExtraMetrics(m)
+	}
+	extraResourceMetrics.MoveAndAppendTo(rms)
+
 	for i := 0; i < rms.Len(); i++ {
```

Comment:
It seems like it would be easy enough to make this a function (e.g. pushResourceMetrics(ctx context.Context, rms pmetric.ResourceMetrics) error), and then call it twice: Once for actual metrics, and once for target_info.  Then we could avoid the copy.

Reply:
Keeping it in a single call feels cleaner to me than adding a special path. This way whether you're debugging normal exporter or GMP, the logic is the same. I did make your change to avoid unnecessary copies as you suggested below though, so ptal